components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OAS2Parser.java

```diff
@@ -383,10 +383,27 @@ public Set<URITemplate> getURITemplates(String resourceConfigsJSON) throws APIMa
                         template.setAuthType("Any");
                         template.setAuthTypes("Any");
                     }
-                    if (extensions.containsKey(APIConstants.SWAGGER_X_THROTTLING_TIER)) {
+                    if (extensions.containsKey(APIConstants.SWAGGER_X_THROTTLING_TIER) &&
+                            !extensions.get(APIConstants.SWAGGER_X_THROTTLING_TIER).toString().isEmpty()) {
```

Comment:
shall we use StringUtils.isNotBlank instead? 
```suggestion
                    if (StringUtils.isNotBlank(extensions.get(APIConstants.SWAGGER_X_THROTTLING_TIER))) {
```

Reply:
Please do the same for other instances as well where we have the isEmpty() check